akka-diagnostics/src/main/scala/akka/diagnostics/ConfigChecker.scala

```diff
@@ -555,6 +556,29 @@ class ConfigChecker(system: ExtendedActorSystem, config: Config, reference: Conf
       else Nil
     }
 
+  private def checkInternalDispatcherSize(): List[ConfigWarning] =
+    ifEnabled("internal-dispatcher-size") { checkerKey =>
+      val path = internalDispatcherPath
+
+      val size = dispatcherPoolSize(config.getConfig(path))
+
+      val availableProcessors = Runtime.getRuntime.availableProcessors
+      if (size > 64 && size > availableProcessors)
+        warn(
+          checkerKey,
+          path,
+          s"Don't use too large pool size [$size] for the internal-dispatcher. " +
+          "Note that the pool size is calculated by ceil(available processors * parallelism-factor), " +
+          "and then bounded by the parallelism-min and parallelism-max values. " +
+          s"This machine has [$availableProcessors] available processors. " +
+          "If you use a large pool size here because of blocking execution you should instead use " +
+          "a dedicated dispatcher to manage blocking tasks/actors. Blocking execution shouldn't " +
+          "run on the internal-dispatcher because that may starve other tasks.")
```

Comment:
No user code should run on the internal-dispatcher (possibly the serializers?), it's there to isolate Akka internal stuff from user stuff, so the part about "don't run blocking stuff on it" is maybe a bit weird. The user shouldn't use it for running _anything_, so should it say that instead?

Reply:
good point. changed.

Reply:
done